### ISSUE TITLE
Incorrect indent when methods are chained to inline new instance

### ISSUE THREAD

**Issue by paustint** [dangmai/prettier-plugin-apex#435]
_Sorry for the cryptic title!_

```sh
# Prettier options (if any):
{
  "trailingComma": "none",
  "apexInsertFinalNewline": false,
  "singleQuote": true,
  "printWidth": 140,
  "overrides": [
    {
      "files": "**/classes/**/*.cls",
      "options": { "tabWidth": 4 }
    },
    {
      "files": "**/lwc/**/*.html",
      "options": { "parser": "lwc", "tabWidth": 4 }
    },
    {
      "files": "*.{cmp,page,component}",
      "options": { "parser": "html", "tabWidth": 4 }
    }
  ]
}

```

**Input:**

```apex
new Flow.Interview.Termination_Handle_Legal_Comments_and_Chatter_Posts(new Map<String, Object>{ 'varCaseId' => currCase.Id, 'varAccountId' => currCase.AccountId, 'varType' => chatterType }).start();
```

**Actual output:**

```apex
new Flow.Interview.Termination_Handle_Legal_Comments_and_Chatter_Posts(
        new Map<String, Object>{ 'varCaseId' => currCase.Id, 'varAccountId' => currCase.AccountId, 'varType' => chatterType }
    )
    .start();
```

**Expected output:**

```apex
// prettier-ignore
new Flow.Interview.Termination_Handle_Legal_Comments_and_Chatter_Posts(
        new Map<String, Object>{ 'varCaseId' => currCase.Id, 'varAccountId' => currCase.AccountId, 'varType' => chatterType }
)
.start();
```

**Additional information (please fill this out)**:

- OS: OSX
- Version: `"prettier-plugin-apex": "^1.10.0"`

**Comment by dangmai** [comment]
Thanks for the issue! This is indeed a difficult one to write a title for 🤣 I'll take a look at this soon-ish, right now I'm rewriting this library in TypeScript so formatting changes need to take a bit of a back seat, but hopefully that won't take too long.

**Comment by paustint** [comment]
@dangmai - No problem! I just wanted to let you know about the bug with no expectations beyond that. You are an excellent maintainer and THANK YOU for this library and the ongoing work you put into it.

As for Typescript, that is awesome! The only JavaScript I write these days is for Salesforce, but aside from that I am 100% sold on using TypeScript for everything because it is so amazing. 😍

**Comment by dangmai** [comment]
Looking at this issue now, there's actually some complexity to it. Your expected output has `.start()` on the same level as the parent expression, which makes it difficult to understand that it's a child expression. I would actually rather go with how Prettier upstream (for JS) handles this kind of complexity, example [here](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEU4HcAEAJOAbPCAdQgCc8ATACgDc5SBPQqAcwDUBDUgSw4CM8cAHIcAtnAAqACw4wi3AgGUArqXwMAQmo4BrDXDAdlAZzgB5AGbYI6NvT4RTAQSgUAwqQ4AvBgEkYvsYAlAB0dIwAMtAsALJwMFIQ7hwE0rJuHFAqania2jrScAAKPLDyFAlmUP7GZuhQ8gkQyjAuDABiqgn0cU3JBFRBANwAOlBjaFi4BMRklLT0TNGcPPyCIuJpcgp42epacLr6hibmVjZ2pA7Orh7efgHBYYtRrL2J-XhbGVmq+-mFErcMrcCpSKo1OoNbhNFptTqkbqkd5JDIDULGGBcGAAJkGQxAABoQBAAA4wbjQYzIUBcUg2IpcBDUlApdAcBjU4l8TxgHTxRRiOARYFwZAWFKmbm8-kwRSkjhgYEsZAwUjKODEuCiPhwCgUPURTIsZQcFhwdpkUSyCmsZAgIwwCBEkBSGCiPBEKQwuDGBVgOCKZkw7g0GEMe1gYxckDA0yIkpm63iyWakAAK2MAA9FMrBABFZQQeApvBSkAK0jx+18fj4F2k0rbMHIAAcAAZiY3HHAiJ5SfbG776HQXQBHIvwEpklkO4wAWjQer1LrUE+4akTLGTSAlZbTplE3FV6oPebgheLYt3qeJWL45QSyBxd88CmVbggoh3IF9AFYXVOCR+BZPdyxoDVfFcBA5TAHhyRcChFBgJhrzAuAAF8MKAA).

Basically, if there's only 1 method call, it's always attached directly to the parent expression with no line break, no matter how long it is. If there are more than 1, new lines with another level of indentation is added.

Going down this route will also make it less of a cognitive switch when looking at formatted JS code and formatted Apex code, because they will have the same formatting behavior.
